packages/components/progress-tracker/src/ProgressTrackerStep.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  type ComponentPropsWithoutRef,
+  forwardRef,
+  type ReactNode,
+  useEffect,
+  useId,
+  useMemo,
+  useState,
+} from 'react'
+
+import {
+  ProgressTrackerStepContext,
+  type ProgressTrackerStepContextInterface,
+  useProgressTrackerContext,
+} from './ProgressTrackerContext'
+import { stepButtonVariant, stepItemVariant } from './ProgressTrackerStep.styles'
+import { ProgressTrackerStepIndicator } from './ProgressTrackerStepIndicator'
+
+export type ProgressTrackerStepProps = ComponentPropsWithoutRef<'li'> &
+  (
+    | {
+        disabled?: boolean
+        children: ReactNode
+      }
+    | {
+        disabled?: boolean
+        'aria-label': string
+      }
+  )
+
+export const ProgressTrackerStep = forwardRef<HTMLLIElement, ProgressTrackerStepProps>(
+  ({ disabled = false, children, 'aria-label': ariaLabel, className, ...rest }, ref) => {
+    const {
+      stepIndex: activeStepIndex,
+      steps,
+      onStepClick,
+      setSteps,
+      size,
+      readOnly,
+    } = useProgressTrackerContext()
+
+    const ID = useId()
+    const stepId = `step-${ID}`
+    const stepIndex = [...steps.keys()].indexOf(stepId)
+
+    const disabledAfter = useMemo(() => {
+      const nextStepId = [...steps.keys()][stepIndex + 1]
+
+      return !!(nextStepId && steps.get(nextStepId)?.includes('disabled'))
+    }, [steps, stepIndex])
+
+    const [progressState, setProgressState] =
```

Comment:
I think we can do without this `useState`, i.e.:
```tsx
progressState = (() => {
  if (stepIndex === activeStepIndex) 'active'
  else if (stepIndex < activeStepIndex) return 'complete'
  return 'incomplete'
})()
```

Reply:
Thx, I'll test this option :+1: But to be curious, what would be the real gain of this method?

Reply:
https://react.dev/learn/you-might-not-need-an-effect#chains-of-computations

I would flip this question around, and ask you what's the benefit of using `useState` and `useEffect` when you can get by without them 😉

Reply:
I knew you would send this link...